### PR TITLE
fix(core): allow required nil values

### PR DESCRIPTION
**Pull request by Sh4d1**
Signed-off-by: Patrik Cyvoct <pcyvoct@scaleway.com>

### CODE REVIEW

File: internal/core/validate.go [hunk 1]
```diff
@@ -72,9 +77,9 @@ func validateRequiredArgs(cmd *Command, cmdArgs interface{}) error {
 			panic(validationErr)
 		}
 
-		for _, fieldValue := range fieldValues {
-			if arg.Required && (fieldValue.IsZero() || !fieldValue.IsValid()) {
-				return MissingRequiredArgumentError(arg.Name)
+		for i := range fieldValues {
+			if !rawArgs.ExistsArgByName(strings.Replace(arg.Name, "{index}", strconv.Itoa(i), 1)) {
```
Comment (jerome-quere) [inline internal/core/validate.go:81]: shoud be `0`

File: internal/args/args.go [hunk 2]
```diff
@@ -15,6 +15,13 @@ const emptySliceValue = "none"
 // RawArgs allows to retrieve a simple []string using UnmarshalStruct()
 type RawArgs []string
 
+// ExistsArgByName checks if the given argument exists in the raw args
+func (a RawArgs) ExistsArgByName(name string) bool {
```
[the remark this answers is not in the document]
Comment (Sh4d1) [reply internal/args/args.go:19]: Hum what would be the usecase of the returned string?

File: internal/core/validate.go [hunk 3]
```diff
@@ -59,8 +60,12 @@ func validateArgValues(cmd *Command, cmdArgs interface{}) error {
...
 // Returns an error for the first missing required arg.
 // Returns nil otherwise.
-func validateRequiredArgs(cmd *Command, cmdArgs interface{}) error {
+func validateRequiredArgs(cmd *Command, cmdArgs interface{}, rawArgs args.RawArgs) error {
 	for _, arg := range cmd.ArgSpecs {
+		if !arg.Required {
+			continue
+		}
+
 		fieldName := strcase.ToPublicGoName(arg.Name)
```
[the remark this answers is not in the document]
Comment (Sh4d1) [reply internal/core/validate.go:69]: We still need to get the reflect values though no?

File: internal/core/validate.go (hunk 1 above)
Comment (Sh4d1) [reply internal/core/validate.go:81]: why 0? :thinking:

**Pull request merged**